lib/internal/per_context/primordials.js

```diff
@@ -402,60 +396,30 @@ const makeSafe = (unsafe, safe) => {
   ObjectFreeze(safe);
   return safe;
 };
+
 primordials.makeSafe = makeSafe;
 
 // Subclass the constructors because we need to use their prototype
 // methods later.
 // Defining the `constructor` is necessary here to avoid the default
 // constructor which uses the user-mutable `%ArrayIteratorPrototype%.next`.
-primordials.SafeMap = makeSafe(
-  Map,
-  class SafeMap extends Map {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-primordials.SafeWeakMap = makeSafe(
-  WeakMap,
-  class SafeWeakMap extends WeakMap {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-
-primordials.SafeSet = makeSafe(
-  Set,
-  class SafeSet extends Set {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-primordials.SafeWeakSet = makeSafe(
-  WeakSet,
-  class SafeWeakSet extends WeakSet {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-
-primordials.SafeFinalizationRegistry = makeSafe(
-  FinalizationRegistry,
-  class SafeFinalizationRegistry extends FinalizationRegistry {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(cleanupCallback) { super(cleanupCallback); }
-  },
-);
-primordials.SafeWeakRef = makeSafe(
-  WeakRef,
-  class SafeWeakRef extends WeakRef {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(target) { super(target); }
-  },
-);
+[
+  'Map',
+  'WeakMap',
+  'Set',
+  'WeakSet',
+  'FinalizationRegistry',
+  'WeakRef',
+  'Promise',
+].forEach((unsafeClassName) => {
+  // eslint-disable-next-line no-restricted-globals
+  const unsafeClass = globalThis[unsafeClassName];
+  primordials['Safe' + unsafeClassName] = makeSafe(unsafeClass, class extends unsafeClass {});
+});
 
-const SafePromise = makeSafe(
-  Promise,
-  class SafePromise extends Promise {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(executor) { super(executor); }
-  },
-);
+const {
+  SafePromise,
+} = primordials;
```

Comment:
Note that `SafePromise` used to not be on the `primordials` object, it was only used locally on that file; it probably makes sense to keep it that way.